### PR TITLE
Include language tour

### PR DESCRIPTION
**Pull request by b-studios**
Please see the sibling PR for a detailed discussion: https://github.com/effekt-lang/effekt/pull/536

### CODE REVIEW

File: _data/menu.yml [hunk 1]
```diff
@@ -4,10 +4,44 @@ options:
...
+        url: docs/tutorial/effects
+      - title: Objects
+        url: docs/tutorial/objects
+      - title: Captures
+        url: docs/tutorial/captures
+      - title: Namespaces
+        url: docs/tutorial/namespaces
+      - title: Regions
+        url: docs/tutorial/regions
+      - title: IO
```
Comment (b-studios) [inline _data/menu.yml:41]:
```suggestion
      - title: File IO
```
since we are currently only discusssing this

File: docs/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+layout: docs
+title: Tutorial
```
Comment (b-studios) [inline docs/tutorial.md:3]:
```suggestion
title: Language Tour
```

same for the file name and the folder (which should be `tour`).

File: docs/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+position: 3
+permalink: docs/tutorial
```
Comment (b-studios) [inline docs/tutorial.md:5]:
```suggestion
permalink: tour
```
would be nice if this would just be `effekt-lang.org/tour`

**Review by b-studios — COMMENTED** [review]
@dvdvgt could you update the compiler in this branch, so that we can check the new tour on IO?

File: _includes/install.md [hunk 4]
```diff
@@ -1,17 +1,23 @@
+#### Requirements
+
 Currently, the Effekt language is implemented in Scala and compiles to JavaScript (besides other backends).
 To use the Effekt compiler and run the resulting programs, you need to have
 
-- Java (>= 11),
-- [Node.js](https://nodejs.org/en/) (>= 12), and
+- Java (>= 17),
```
Comment (jiribenes) [inline _includes/install.md:6]: Why? :)
We're still supporting Java 11, aren't we?

File: _includes/install.md [hunk 5]
```diff
@@ -20,17 +26,24 @@ And then install it with
 npm install -g <PATH_TO_FILE>/effekt.tgz
 ```
 
+#### Install location
+
 You can find the installation location by running
 
 ```bash
-npm root -g
+$ npm config get prefix
```
Comment (jiribenes) [inline _includes/install.md:34]: the `$` is inconsistent now with the rest of the `bash` excerpts

Comment (dvdvgt) [reply _includes/install.md:34]: I have Effekt install via `sbt install` which in turn uses `npm install -g`, yet

```
$ npm root -g
/opt/homebrew/lib/node_modules

$ npm config get prefix
/opt/homebrew
```

while

```
$ which effekt
/opt/homebrew/bin/effekt
```

Hence, this should work in any case by just including more directories in the `$PATH`.

Comment (jiribenes) [reply _includes/install.md:34]: The file in `bin` is just a symlink, but the bigger problem is that the `npm root -g` can be user configured to be a different folder outside of the npm prefix.

Comment (dvdvgt) [reply _includes/install.md:34]: Good point, thanks. I'll change it again.

> [...] but the bigger problem is that the npm root -g can be user configured to be a different folder outside of the npm prefix.

Well, we cannot account for every eventuality.

File: _includes/install.md (hunk 4 above)
Comment (b-studios) [reply _includes/install.md:6]: Yes, you are right

https://github.com/effekt-lang/effekt/blob/master/.github/workflows/ci.yml#L10

I thought we updated that even in the CI.

**Comment by jiribenes** [comment]
Resolves https://github.com/effekt-lang/effekt-website/issues/63.

File: _data/menu.yml (hunk 1 above)
Comment (dvdvgt) [reply _data/menu.yml:41]: In the Effekt repo this file has the title `Asynchronous IO (Input-Output)`. I'd just leave it like this to not have two different names. Also, we will probably expand this section with other parts of IO in the future.

File: tour/adts.effekt.md [hunk 6]
```diff
@@ -0,0 +1 @@
+../effekt/examples/features/adts.effekt.md
```
Comment (jiribenes) [inline tour/adts.effekt.md:1]: Just noticed this: Does this mean that the `adt.effekt.md` with a checkfile (singular) isn't linked here?
If so, what's the purpose of that file in `effekt/examples/features`? 🤔

File: tour/target
[the remark this answers is not in the document]
Comment (dvdvgt) [reply tour/target:1]: Some folder I added by accident since it is not in the .gitignore. Good catch, thanks

File: tour/adts.effekt.md (hunk 6 above)
Comment (dvdvgt) [reply tour/adts.effekt.md:1]: When deleting it, we should probably also rename the folder from `features` to `tour`

**Review by jiribenes — APPROVED** [review]
Ship it :shipit:
We can resolve the details later.

**Pull request merged**